### PR TITLE
Rewrite printPreviewDialog, label and DataGridView MAUI tests

### LINKED ISSUE #3617: MAUI tests fail frequently and aren't easily diagnosable
These are an attempt to shim in the old internal tests (note that MAUI here is an old internal test framework). The old tests worked quite differently and don't fit in well with the current test infrastructure. We're getting [frequent intermittent failures](https://dev.azure.com/dnceng/public/_build/results?buildId=737706&view=ms.vss-test-web.build-test-results-tab&runId=22844842&resultId=100000&paneView=history) like so:

```
System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(scenarioName: \"Hotkey_Fires_OnClick\")

Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Pass\r\nActual:   Fail\r\n          ↑ (pos 0)

Stack trace
   at System.Windows.Forms.Maui.IntegrationTests.MauiTestHelper.ValidateScenarioPassed(String projectName, String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/MauiTestHelper.cs:line 90
   at System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiButtonTests.cs:line 35
``` 

This doesn't really help you much and it isn't obvious as to what you can do to move forward. In addition, the MAUI infrastructure sources aren't available so it is difficult to understand how the whole system works.

My recommendation (and current POR):

1. Disable the MAUI tests for now
2. Make them more xUnit-like and have any shim code in the repo
3. If we can make the tests plug in as a "normal" xUnit test then re-enable them for the CI, otherwise leave them for manual runs

cc: @AdamYoblick, @RussKie 

Failed with #3600

### PR DESCRIPTION
**Pull request by Danil-Andrianov**
Fixes #3617

## Proposed changes

- Replaced DataGridViewMAUI tests into UI IntegrationTests project
- Replaced LabelMAUI tests into UI IntegrationTests project
- Replaced PrintPreviewDialogMAUI tests into UI IntegrationTests project

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6262)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            await RunTestAsync((form, label) =>
+            {
+                label.AutoSize = false;
+
+                label.Size = new Size(10, 10);
+                label.Text = "Hello";
+                Size oldSize = label.Size;
+                label.Text = "Say Hello";
+                Size newSize = label.Size;
+                Assert.Equal(newSize, oldSize);
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs:30]: Add a blank line before the Assert here and for the rest test as well

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/PrintPreviewDialogTests.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        public async Task Hotkey_Ctrl_1Async()
+        {
+            await RunTestAsync((form, textBox) =>
+            {
+                PrintPreviewDialog printPreviewDialog = new();
+                printPreviewDialog.Show();
+                SendKeys.SendWait("^1");
+
+                Assert.Equal(1, printPreviewDialog.PrintPreviewControl.Rows);
+
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/PrintPreviewDialogTests.cs:27]: Remove blank lines between Asserts

**Review by RussKie — CHANGES_REQUESTED** [review]

**Comment by RussKie** [comment]
Please review this doc as well: https://github.com/dotnet/winforms/blob/main/docs/testing.md and correct the naming as per https://github.com/dotnet/winforms/blob/main/docs/testing.md#naming section.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+using System.Drawing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Windows.Forms.UITests
+{
+    public class LabelTests : ControlTestBase
+    {
+        public LabelTests(ITestOutputHelper testOutputHelper)
```
[the remark this answers is not in the document]
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs:13]: `AutoSize` property doesn't work correctly without the form, therefore I suggest to remain things as it is.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+{
+    public class DataGridViewTests : ControlTestBase
+    {
+        public DataGridViewTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        // This test is rather specific: there should be a separate Form to reproduce the issue.
+        // We need to show this Form ourselves, the issue won't reproduce with ReflectBase-inherited Form which is shown automatically while scenarios is being run.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:20]: This comment is no longer relevant

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                dataGridView.ShowCellToolTips = true;
+                dataGridView.DataSource = dataTable;
+
+                Point point = dataGridView.GetCellDisplayRectangle(columnIndex: 0, rowIndex: 0, cutOverflow: false).Location;
+
+                // Move mouse cursor over any cell of the first row.
+                await InputSimulator.SendAsync(
+                            form,
+                            inputSimulator => inputSimulator.Mouse.MoveMouseTo(point.X, point.Y));
+                form.Close();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:40]: We need a comment for this line, as it's not generally expected to close the dialog in the body of the test.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        [WinFormsFact]
+        public async Task DataGridView_ToolTip_DoesNot_ThrowExceptionAsync()
+        {
+            await RunTestAsync(async (form, dataGridView) =>
+            {
+                using DataTable dataTable = new();
+                dataTable.Columns.Add(columnName: "name");
+                dataTable.Rows.Add(values: "name1");
+
+                // Create DataGridView with DataSource set to dataTable.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:30]: This comment is largely redundant - it's pretty obvious what it's doing.

**Review by RussKie — COMMENTED** [review]
Almost there :)

**Comment by Danil-Andrianov** [comment]
@RussKie https://github.com/dotnet/winforms/pull/6262/#discussion_r770104067
Do we really needed to have commentary there? What kind of commentary do you expect there?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs (hunk 4 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:20]: Do we really needed in any comment there? What kind of comment do you expect?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/LabelTests.cs]: I made a mistake: those label tests could be moved to the unit tests project if we merely add label to the form. So I just did it.

**Comment by dreddy-work** [comment]
LGTM.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                dataGridView.ShowCellToolTips = true;
+                dataGridView.DataSource = dataTable;
+                Point point = dataGridView.GetCellDisplayRectangle(columnIndex: 0, rowIndex: 0, cutOverflow: false).Location;
+
+                // Move mouse cursor over any cell of the first row.
+                await InputSimulator.SendAsync(
+                            form,
+                            inputSimulator => inputSimulator.Mouse.MoveMouseTo(point.X, point.Y));
+
+                // We need to close the form and then check whether dataTable works fine or it doesn't
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:36]:
```suggestion
                // Close the form to verify no exceptions thrown while showing the tooltip.
                // Regression test for https://github.com/dotnet/winforms/issues/5496
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            await RunTestAsync(async (form, dataGridView) =>
+            {
+                using DataTable dataTable = new();
+                dataTable.Columns.Add(columnName: "name");
+                dataTable.Rows.Add(values: "name1");
+                dataGridView.ShowCellToolTips = true;
+                dataGridView.DataSource = dataTable;
+                Point point = dataGridView.GetCellDisplayRectangle(columnIndex: 0, rowIndex: 0, cutOverflow: false).Location;
+
+                // Move mouse cursor over any cell of the first row.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:31]:
```suggestion
                // Move mouse cursor over any cell of the first row to trigger a tooltip.
```

@vladimir-krestov  @ArtemTatarinov is there a way to check a tooltip is shown?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                try
+                {
+                    dataTable.AcceptChanges();
+                }
+                catch
+                {
+                    exceptionThrown = true;
+                }
+
+                Assert.False(exceptionThrown);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:48]:
```suggestion
                dataTable.AcceptChanges();
```

If there's an exception - the test will fail.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/Infra/ControlTestBase.cs [hunk 10]
```diff
@@ -250,6 +263,37 @@ protected async Task RunFormAsync<T>(Func<(Form dialog, T control)> createDialog
...
+                await gate.Task;
+                await JoinableTaskFactory.SwitchToMainThreadAsync();
+                await WaitForIdleAsync();
+                try
+                {
+                    await testDriverAsync(dialog!);
+                }
+                finally
+                {
+                    dialog!.Close();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/Infra/ControlTestBase.cs:282]: The dialog must be disposed of. Please see `RunFormAsync` above.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 11]
```diff
@@ -283,6 +283,40 @@ public void Label_ImageAlign_Set_GetReturnsExpected(bool autoSize, ContentAlignm
...
+            label.Parent = form;
+            label.AutoSize = true;
+            label.Size = new Size(10, 10);
+            label.Text = "Hello";
+            Size oldSize = label.Size;
+            label.Text = "Say Hello";
+            Size newSize = label.Size;
+
+            Assert.NotEqual(newSize, oldSize);
+        }
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:318]: I think these two can be combined into a theory test, something like
```suggestion
        [WinFormsTheory]
        [InlineData(false, false)]
        [InlineData(true, true)]
        public void Label_AutoSize_BehavesExpected(bool autoSize, bool expected)
        {
            using Form form = new Form();
            using Label label = new()
            {
                AutoSize = autoSize,
                Size = new(10, 10),
                Text = "Hello",
            };
            Size oldSize = label.Size;
            form.Controls.Add(label);
            form.Show();

            label.Text = "Say Hello";
            Size newSize = label.Size;

            Assert.Equal(expected, newSize == oldSize);
        }

```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                dataGridView.ShowCellToolTips = true;
+                dataGridView.DataSource = dataTable;
+                Point point = dataGridView.GetCellDisplayRectangle(columnIndex: 0, rowIndex: 0, cutOverflow: false).Location;
+
+                // Move mouse cursor over any cell of the first row to trigger a tooltip.
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Mouse.MoveMouseTo(point.X, point.Y));
+
+                // Close the form to verify no exceptions thrown while showing the tooltip.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:36]: We're meant to close, right?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/Infra/ControlTestBase.cs [hunk 13]
```diff
@@ -252,6 +252,39 @@ protected async Task RunFormAsync<T>(Func<(Form dialog, T control)> createDialog
             await test.JoinAsync();
         }
 
+        protected async Task RunFormWithoutControlAsync(Func<PrintPreviewDialog> createForm, Func<PrintPreviewDialog, Task> testDriverAsync)
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/Infra/ControlTestBase.cs:255]: No, no, no... This is a general purpose class and we mustn't hardcode any specific types here.
```suggestion
        protected async Task RunFormWithoutControlAsync<TForm>(Func<TForm> createForm, Func<TForm, Task> testDriverAsync) where TForm: Form
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs (hunk 12 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:36]: Could you clarify what do you mean, please?

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:36]: According to the comment we're meant to close the dialog _before_ we accept changes. This what the original test was doing.

Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs:36]: Oh, I hope now I got you.

**Review by RussKie — APPROVED** [review]

**Pull request merged**